xml_struct/src/lib.rs

```diff
@@ -39,8 +43,66 @@ use thiserror::Error;
 pub use xml_struct_derive::*;
 
 /// A data structure which can be serialized as XML content nodes.
+///
+/// # Usage
+///
+/// The following demonstrates end-to-end usage of `XmlSerialize` with both
+/// derived and manual implementations.
+///
+/// ```
+/// use quick_xml::{
+///     events::{BytesText, Event},
+///     writer::Writer
+/// };
+/// use xml_struct::{Error, XmlSerialize};
+///
+/// #[derive(XmlSerialize)]
+/// #[xml_struct(default_ns = "http://foo.example/")]
+/// struct Foo {
+///     some_field: String,
+///
+///     #[xml_struct(flatten)]
+///     something_else: Bar,
+/// }
+///
+/// enum Bar {
+///     Baz,
+///     Foobar(String),
```

Comment:
note (non-blocking): "qux" is the element in the list of metasyntactic variables that follows "baz" (see https://en.wikipedia.org/wiki/Metasyntactic_variable#General_usage)